game/scripts/vscripts/items/sonic.lua

```diff
@@ -0,0 +1,64 @@
+item_sonic = class(ItemBaseClass)
+
+LinkLuaModifier("modifier_item_sonic", "items/sonic.lua", LUA_MODIFIER_MOTION_NONE)
+LinkLuaModifier("modifier_sonic_fly", "modifiers/modifier_sonic_fly.lua", LUA_MODIFIER_MOTION_NONE)
+
+function item_sonic:GetIntrinsicModifierName()
+  return "modifier_sonic_fly"
```

Comment:
Intrinsic modifier is the one that's always on. You want "modifier_item_sonic" here.